pkg/controller/virtualmachineimport/virtualmachineimport_controller.go

```diff
@@ -343,39 +399,308 @@ func (r *ReconcileVirtualMachineImport) addWatchForImportPod(instance *v2vv1.Vir
+			Parallelism:  &parallelism,
+			BackoffLimit: &backoffLimit,
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: virtV2vJobName + "-",
+					Namespace:    vmSpec.Namespace,
+				},
+				Spec: v1.PodSpec{
+					RestartPolicy: v1.RestartPolicyNever,
+					Containers: []v1.Container{
+						{
+							Name:            virtV2vJobName,
+							Image:           virtV2vImage,
+							ImagePullPolicy: v1.PullIfNotPresent,
+							VolumeMounts:    volumeMounts,
+						},
+					},
+					Volumes: volumes,
+				},
+			},
+		},
+		Status: batchv1.JobStatus{},
+	}
+}
+
+func makeJobVolumeMounts(vmSpec *kubevirtv1.VirtualMachine) ([]v1.Volume, []v1.VolumeMount) {
+	volumes := make([]v1.Volume, 0)
+	volumeMounts := make([]v1.VolumeMount, 0)
+	// add volumes and mounts for each of the VM's disks.
+	// the virt-v2v pod expects to see the disks mounted at /mnt/disks/diskX
+	for i, dataVolume := range vmSpec.Spec.Template.Spec.Volumes {
+		vol := v1.Volume{
+			Name: dataVolume.DataVolume.Name,
+			VolumeSource: v1.VolumeSource{
+				PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+					ClaimName: dataVolume.DataVolume.Name,
+					ReadOnly:  false,
+				},
+			},
+		}
+		volumes = append(volumes, vol)
+
+		volMount := v1.VolumeMount{
+			Name:      dataVolume.DataVolume.Name,
+			MountPath: fmt.Sprintf("/mnt/disks/disk%v", i),
+		}
+		volumeMounts = append(volumeMounts, volMount)
+	}
+
+	// add volume and mount for the libvirt domain xml config map.
+	// the virt-v2v pod expects to see the libvirt xml at /mnt/v2v/input.xml
+	volumes = append(volumes, v1.Volume{
+		Name: vmSpec.Name,
+		VolumeSource: v1.VolumeSource{
+			ConfigMap: &v1.ConfigMapVolumeSource{
+				LocalObjectReference: v1.LocalObjectReference{
+					Name: vmSpec.Name,
+				},
+			},
+		},
+	})
+	volumeMounts = append(volumeMounts, v1.VolumeMount{
+		Name:      vmSpec.Name,
+		MountPath: "/mnt/v2v",
+	})
+	return volumes, volumeMounts
+}
+
+// findLibvirtDomainConfigMap finds the libvirt domain xml configmap for a given VM, returning nil if it can't be found.
+func (r *ReconcileVirtualMachineImport) findLibvirtDomainConfigMap(vmName types.NamespacedName) (*v1.ConfigMap, error) {
+	configMapList := &v1.ConfigMapList{}
+	matchingLabels := client.MatchingLabels(map[string]string{VMLabel: vmName.Name})
+	err := r.client.List(context.TODO(), configMapList, matchingLabels)
+	if err != nil {
+		return nil, err
+	}
+	if len(configMapList.Items) > 0 {
+		return &configMapList.Items[0], nil
+	}
+	return nil, nil
+}
+
+// makeLibvirtDomainConfigMap creates a libvirt domain xml configmap for a VM to be used during guest conversion
+func (r *ReconcileVirtualMachineImport) makeLibvirtDomainConfigMap(instance *v2vv1.VirtualMachineImport, vmSpec *kubevirtv1.VirtualMachine) (*v1.ConfigMap, error) {
+	domain := r.makeLibvirtDomain(vmSpec)
+	domxml, err := xml.Marshal(domain)
+	if err != nil {
+		return nil, err
+	}
+	domainXMLConfigMap := &v1.ConfigMap{
```

Comment:
What do you think about using https://github.com/kubevirt/vm-import-operator/blob/master/pkg/configmaps/configmaps.go? Like in https://github.com/kubevirt/vm-import-operator/blob/2cefbcc19ab4c5e98b12a08a18fa978de9cbb282/pkg/providers/ovirt/provider.go#L467